core/variant/variant_parser.cpp

```diff
@@ -1938,14 +1938,18 @@ static String rtos_fix(double p_value, bool p_compat) {
 	if (p_value == 0.0) {
 		return "0"; //avoid negative zero (-0) being written, which may annoy git, svn, etc. for changes when they don't exist.
 	} else if (std::isnan(p_value)) {
-		return "nan";
+		return "NAN";
 	} else if (std::isinf(p_value)) {
 		if (p_value > 0) {
-			return "inf";
+			if (p_compat) {
+				return "inf";
+			} else {
+				return "INF";
+			}
 		} else if (p_compat) {
 			return "inf_neg";
 		} else {
-			return "-inf";
```

Comment:
Unsure what `p_compat` is for, but there's an inconsistency here. `-inf` is not returned at all in this PR, but `inf_neg` is, even though it's older. I don't understand the justification

Reply:
`-INF` is the preferred option for saving new files, `inf_neg` is compatible with all 4.x versions and so is used when `p_compat` is `true`. There is no reason to add an option for `-inf`, since `inf_neg` is compatible with all versions.